src/core/thread/topology.hpp

```diff
@@ -352,6 +352,31 @@ class Neighbor
      */
     void SetRloc16(uint16_t aRloc16) { mRloc16 = aRloc16; }
 
+    /**
+     * This method indicates whether Enhanced Keep-Alive is supported or not.
+     *
+     * @returns TRUE if Enhanced Keep-Alive is supported, FALSE otherwise.
+     *
+     */
+    bool IsEnhancedKeepAliveSupported(void) const
+    {
+        return mState != kStateInvalid && mVersion >= OPENTHREAD_THREAD_VERSION_1_2;
```

Comment:
nit: what about `assert(mState != kStateInvalid);` here

Reply:
@bukepo

Reply:
intended not to use assert() to make sure a Parent is valid.